packages/cli/src/start/server.ts

```diff
@@ -0,0 +1,199 @@
+import fs from 'node:fs'
+import path from 'node:path'
+
+import { type AnyModels, handleGetRequest } from '@rise-tools/server'
+import chokidar from 'chokidar'
+import Fastify from 'fastify'
+import internalIp from 'internal-ip'
+import qrcode from 'qrcode-terminal'
+import { WebSocketServer } from 'ws'
+import z from 'zod'
+
+// TODO
+import { connectWebSocket, createWSServerContext } from '../../../server/src/ws-connection'
+import { IGNORED_PATH, WATCH_PATH } from '../config/constants'
+
+interface IndexedModels {
+  [key: string]: any
+}
+
+export const devArgsSchema = z.object({
+  ws: z.boolean().optional().default(false),
+  prod: z.boolean().optional().default(false),
+  port: z.number(),
+  host: z.string(),
+  cwd: z.string().optional(),
+})
+
+export type DevArgs = z.infer<typeof devArgsSchema>
+
+class ModelManager {
+  models: AnyModels & IndexedModels = {}
+  cwd: string = process.cwd()
+
+  constructor(cwd: string) {
+    this.cwd = cwd
+  }
+
+  parseModelPath(modelPath: string) {
+    const key = modelPath.split('/').slice(1, -1).join(':')
+    const path = modelPath.split('.').slice(0, -1).join('.')
+    return { key, path }
+  }
+
+  async updateModel(modelPath: string) {
+    const { key, path: basePath } = this.parseModelPath(modelPath)
+    const cwd = this.cwd!
+    this.models[key] = await import(path.join(cwd, basePath))
+  }
+
+  removeModel(modelPath: string) {
+    const { key } = this.parseModelPath(modelPath)
+    this.models[key]
+  }
+
+  getModels() {
+    return this.models
+  }
+}
+
+export class DevServer {
```

Comment:
I'm not sure why we need to use a class for this. Because this is a CLI, there will only ever be one instance.

I would like to avoid the complexity of classes, if possible. All the properties here can just be module scoped variables.

Same for ModelManager.

If we do want to support multiple dev servers, we could just use a function. Something like this:

```ts
function createDevServer() {
  let internalState = ...
  function start() { ...  }
  return { start }
}
```

We can keep the classes for now if you feel strong about them. But if Mike or I wrote this code, we would just use module state or functions. (See how the rise servers just use functions rather than classes)